src/gtfs.py

```diff
@@ -74,50 +74,134 @@ def trips_by_route_id(self, route_id: str):
 @tracer.wrap()
 def _download_gtfs_archives_list() -> pd.DataFrame:
     """Downloads list of GTFS archive urls. This file will get overwritten."""
-    archives_df = pd.read_csv(urljoin(GTFS_ARCHIVES_PREFIX, GTFS_ARCHIVES_FILENAME))
-    archives_df.to_csv(MAIN_DIR / GTFS_ARCHIVES_FILENAME)
-    return archives_df
+    try:
+        archives_df = pd.read_csv(urljoin(GTFS_ARCHIVES_PREFIX, GTFS_ARCHIVES_FILENAME))
+        archives_df.to_csv(MAIN_DIR / GTFS_ARCHIVES_FILENAME)
+        return archives_df
+    except (PermissionError, OSError, IOError) as e:
+        logger.error(f"Failed to write GTFS archives file due to permission error: {e}")
+        logger.warning("Continuing with downloaded archives data without saving to disk")
+        return archives_df
```

Comment:
Risk that `archives_df` isn't defined if an exception is throw on L78?

Reply:
Good point, I initialized it above first just in case